nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -140,38 +127,59 @@ define(['StagingServiceClient', 'common/html', 'common/runtime', 'common/ui', 'u
             this.cssBaseClass = cssBaseClass;
         }
 
+        _setUpModel() {
+            // fill in the options for the types
+            const typeArray = Object.keys(this.model.getItem('params'));
+            xsvAppSpec.parameters[0].dropdown_options.options = typeArray.sort().map((opt) => {
+                return {
+                    display: this.fileTypeMapping[opt],
+                    value: opt,
+                };
+            });
+            xsvAppSpec.parameters[0].default_values = typeArray;
+            this.xsvAppSpec = xsvAppSpec;
+            this.spec = Spec.make({ appSpec: this.xsvAppSpec });
+
+            const internalModelContext = {
+                app: {
+                    spec: this.spec,
+                },
+                params: this.spec.makeDefaultedModel(),
+                inputs: {
+                    otherParamIds: this.xsvAppSpec.parameters.map((param) => {
+                        return param.id;
+                    }),
+                },
+            };
+
+            this.internalModel = Props.make({
+                data: internalModelContext,
+            });
+        }
+
         async run() {
+            this._setUpModel();
             const body = this.renderLayout();
-            const formValues = {};
+            this.startInputWidgets();
+            const doThisFirst = function (mdn) {
+                // attach the form widget to the UI modal
+                mdn.querySelector('#' + this.id).appendChild(this.paramsNode);
+            };
+
             const modalArgs = {
                 title: 'Create Import Specification Template',
                 body,
                 okLabel: 'Generate template!',
+                doThisFirst: doThisFirst.bind(this),
                 onConfirm: () => {
-                    // TODO: the form should monitor its state of validity and
-                    // only allow the query to be submitted when the form is valid
-                    const form = document.querySelector('#' + this.id);
-                    const params = ['output_directory', 'output_file_type', 'types'];
```

Comment:
All this is now unnecessary as the values and state of the form are stored in `this.internalModel`, as if this were an app cell.